### PR TITLE
Add forgot password screen

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: lib/auth/screens/reset_password/notifier/reset_password_notifier.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+      state = const Loading();
+
+      await authService.recoverPassword(model);
+
+      state = const Input();
+
+      Navigator.popUntil(
+        routerKey.currentContext!,
+            (route) => route.isFirst == true,
```
Comment (eli1stark) [inline lib/auth/screens/reset_password/notifier/reset_password_notifier.dart:46]: formating issues

File: lib/auth/screens/reset_password/provider/reset_password_notipod.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    final credentialsState = ref.watch(credentialsNotipod);
+    final credentialsNotifier = ref.watch(credentialsNotipod.notifier);
+    final authService = ref.watch(authServicePod);
+    final routerKey = ref.watch(routerKeyPod);
+
+    return ResetPasswordNotifier(
+      credentialsState: credentialsState,
+      credentialsNotifier: credentialsNotifier,
+      authService: authService,
+        routerKey: routerKey,
```
Comment (eli1stark) [inline lib/auth/screens/reset_password/provider/reset_password_notipod.dart:21]: formatting issues

File: lib/auth/screens/reset_password/view/reset_password.dart [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        child: forgotPassword.when(
+          input: (_, __) {
+            return SafeArea(
+              child: Form(
+                child: Padding(
+                  padding: const EdgeInsets.all(24),
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: <Widget>[
+                      const Text(
```
Comment (eli1stark) [inline lib/auth/screens/reset_password/view/reset_password.dart:51]: Repeated logic -> Number 1

File: lib/auth/screens/forgot_password/view/forgot_password.dart [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        child: forgotPassword.when(
+          input: (_, __) {
+            return SafeArea(
+              child: Form(
+                child: Padding(
+                  padding: const EdgeInsets.all(24),
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: <Widget>[
+                      const Text(
```
Comment (eli1stark) [inline lib/auth/screens/forgot_password/view/forgot_password.dart:49]: Repeated logic -> Number 1

File: lib/service/services/authentication/service/helpers/handle_auth_response.dart [hunk 5]
```diff
@@ -7,7 +7,21 @@ AuthenticationModel handleAuthResponse(Map<String, dynamic> json) {
...
   if (result == 0) {
     return response.authModel!;
-  } else if (result == -9) {
+  } else {
+   _validateResultResponse(result);
+   throw '';
+  }
+}
+
+void handleAuthResultResponse(Map<String, dynamic> json) {
```
Comment (eli1stark) [inline lib/service/services/authentication/service/helpers/handle_auth_response.dart:16]: Rename to `handleAuthResult`

File: lib/service/services/authentication/service/helpers/handle_auth_response.dart [hunk 6]
```diff
@@ -7,7 +7,21 @@ AuthenticationModel handleAuthResponse(Map<String, dynamic> json) {
...
+  final result = json['result'] as int;
```
Comment (eli1stark) [inline lib/service/services/authentication/service/helpers/handle_auth_response.dart:17]: add space between lines

File: lib/service/services/authentication/service/helpers/handle_auth_response.dart [hunk 7]
```diff
@@ -7,7 +7,21 @@ AuthenticationModel handleAuthResponse(Map<String, dynamic> json) {
 
   if (result == 0) {
     return response.authModel!;
-  } else if (result == -9) {
+  } else {
+   _validateResultResponse(result);
+   throw '';
```
Comment (eli1stark) [inline lib/service/services/authentication/service/helpers/handle_auth_response.dart:12]: ` throw 'Something went wrong'`; instead of empty throw

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/auth/screens/reset_password/notifier/reset_password_notifier.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/auth/screens/reset_password/notifier/reset_password_notifier.dart:46]: formated

File: lib/auth/screens/reset_password/provider/reset_password_notipod.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/auth/screens/reset_password/provider/reset_password_notipod.dart:21]: formated

File: lib/auth/screens/reset_password/view/reset_password.dart (hunk 3 above)
Comment (VovaMedvediev) [reply lib/auth/screens/reset_password/view/reset_password.dart:51]: refactored

File: lib/auth/screens/forgot_password/view/forgot_password.dart (hunk 4 above)
Comment (VovaMedvediev) [reply lib/auth/screens/forgot_password/view/forgot_password.dart:49]: refactored

File: lib/service/services/authentication/service/helpers/handle_auth_response.dart (hunk 5 above)
Comment (VovaMedvediev) [reply lib/service/services/authentication/service/helpers/handle_auth_response.dart:16]: renamed

File: lib/service/services/authentication/service/helpers/handle_auth_response.dart (hunk 6 above)
Comment (VovaMedvediev) [reply lib/service/services/authentication/service/helpers/handle_auth_response.dart:17]: space added

File: lib/service/services/authentication/service/helpers/handle_auth_response.dart (hunk 7 above)
Comment (VovaMedvediev) [reply lib/service/services/authentication/service/helpers/handle_auth_response.dart:12]: will be discussed in DM

**Pull request merged**